js/src/views/map/map.js

```diff
@@ -18,48 +21,120 @@ var MapView = Backbone.View.extend({
-        sql: 'SELECT * FROM buildings_height',
-        cartocss: '#buildings_height{  marker-fill-opacity: 0.9;  marker-line-color: #FFF;  marker-line-width: 1;  marker-line-opacity: 1;  marker-placement: point;  marker-multi-policy: largest;  marker-type: ellipse;  marker-fill: #FF5C00;  marker-allow-overlap: true;  marker-clip: false;}#buildings_height [ height <= 829.8] {   marker-width: 25.0;}#buildings_height [ height <= 529.5] {   marker-width: 23.3;}#buildings_height [ height <= 392.5] {   marker-width: 21.7;}#buildings_height [ height <= 348] {   marker-width: 20.0;}#buildings_height [ height <= 257.5] {   marker-width: 18.3;}#buildings_height [ height <= 176.65] {   marker-width: 16.7;}#buildings_height [ height <= 164.5] {   marker-width: 15.0;}#buildings_height [ height <= 151.2] {   marker-width: 13.3;}#buildings_height [ height <= 130] {   marker-width: 11.7;}#buildings_height [ height <= 114.3] {   marker-width: 10.0;}',
-        interactivity: 'city, built_name, height'
-      }]
+      css: ''
     }
   },
 
   initialize: function() {
+    this.status = status;
+    this._setListeners();
+  },
+
+  show: function() {
+    this._setView();
+    this._initMap();
+  },
 
-    enquire.register("screen and (max-width:769px)", {
+  _setView: function() {
+    enquire.register("screen and (max-width:767px)", {
       match: _.bind(function(){
         this.mobile = true;
       },this)
     });
 
-    enquire.register("screen and (min-width:770px)", {
+    enquire.register("screen and (min-width:768px)", {
       match: _.bind(function(){
         this.mobile = false;
       },this)
     });
+  },
 
+  _setListeners: function() {
+    this.status.on('change:layer', _.bind(this._activeLayer, this))
   },
 
   _initMap: function() {
     /* this is the definition for basemap */
-    var layer = L.tileLayer('http://{s}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png', {
+    var baseMap = L.tileLayer(this.options.basemap, {
       attribution: '<a href="https://www.mapzen.com/rights">Attribution.</a>. Data &copy;<a href="https://openstreetmap.org/copyright">OSM</a> contributors.'
     });
     /* Here we create the map with Leafleft... */
     this.map = L.map(this.el, this.options.map);
     /* ...and we add the basemap layer with Leaflet as well */
-    this.map.addLayer(layer);
+    this.map.addLayer(baseMap);
 
     return this;
   },
 
-  show: function() {
-    this._initMap();
+  _activeLayer: function() {
+    this._createLayer().done(_.bind(function() {
+      //We remove the previous layer just when the new one arrive.
+      //This way, we are sure we only have one layer at a time.
+
+      this._addLayer();
+    }, this));
+  },
+
+  _createLayer: function() {
+    var sql = this._getLayerQuery();
+    var cartoAccount = this.options.cartodb.user_name;
+    var cartoCss = this.options.cartodb.css || '#table_score_test{ polygon-fill: #FF6600; polygon-opacity: 0.7; line-color: #FFF; line-width: 0.5; line-opacity: 1; }';
```

Comment:
I would move to a options object o CONFIG JSON, if you start to change map params, you have to check the whole view and that can be a little annoying.